x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileDomainIntegTests.java

```diff
@@ -249,14 +259,137 @@ public void testGetProfileByAuthenticationWillNotCheckRealmNameForFileOrNativeRe
+                    // The realm is file if no domain is configured or it can be either file or index if domain is configured
+                    final RealmConfig.RealmIdentifier realmIdentifier = realmDomain == null
+                        ? new RealmConfig.RealmIdentifier("file", "file")
+                        : randomFrom(realmDomain.realms());
+
+                    final Authentication authentication = Authentication.newRealmAuthentication(
+                        new User(username),
+                        new Authentication.RealmRef(
+                            realmIdentifier.getName(),
+                            realmIdentifier.getType(),
+                            randomAlphaOfLengthBetween(3, 8),
+                            realmDomain
+                        )
+                    );
+                    final ProfileService profileService = getInstanceFromRandomNode(ProfileService.class);
+                    final PlainActionFuture<Profile> future = new PlainActionFuture<>();
+                    profileService.activateProfile(authentication, future);
+                    readyLatch.countDown();
+                    startLatch.await();
+                    try {
+                        final String uid = future.actionGet().uid();
+                        logger.info("create profile [{}] for authentication [{}]", uid, authentication);
+                        allUids.add(uid);
+                    } catch (VersionConflictEngineException e) {
+                        // Updating existing profile can error with version conflict. This is the current way
+                        // of handling racing in updating existing profile.
+                        // For this test, it is acceptable to either get the same profile document
+                        // or getting a version conflict for optimistic control (NOT document already exists)
+                        assertThat(e.getMessage(), containsString("version conflict, required seqNo"));
+                    }
+                } catch (Exception e) {
+                    logger.error(e);
+                    fail("caught error when creating new profile: " + e);
+                }
+            });
+            threads[i].start();
+        }
+        if (readyLatch.await(20, TimeUnit.SECONDS)) {
+            startLatch.countDown();
+            for (Thread thread : threads) {
+                thread.join();
+            }
+            // Exactly one profile is created
+            assertThat(allUids, hasSize(1));
+            final String uid = allUids.iterator().next();
+            final Profile profile1 = getProfile(uid, Set.of());
+            assertThat(profile1.uid(), equalTo(uid));
+            assertThat(profile1.user().username(), equalTo(username));
+        } else {
+            fail("Not all threads are ready after waiting");
+        }
+    }
+
+    public void testDifferentiator() {
+        String lastUid = null;
+        for (int i = 0; i < 10; i++) {
```

Comment:
No point in using a `10` size loop IMO, one activate after the `username` change is enough.

Reply:
The loop is to test the erroring behaviour of `differentiator value is too high` which only happens when there are 10 collisions.